static/app/components/projects/bookmarkStar.tsx

```diff
@@ -13,59 +13,67 @@ type Props = {
   organization: Organization;
   project: Project;
   className?: string;
-  /* used to override when under local state */
+  /**
+   * Allows the bookmarked state of the project to be overriden. Useful for
+   * optimistic updates.
+   */
   isBookmarked?: boolean;
   onToggle?: (isBookmarked: boolean) => void;
 };
 
-const BookmarkStar = ({
-  isBookmarked: isBookmarkedProp,
-  className,
-  organization,
-  project,
-  onToggle,
-}: Props) => {
-  const api = useApi();
+const BookmarkStar = styled(
+  ({
+    isBookmarked: isBookmarkedProp,
+    className,
+    organization,
+    project,
+    onToggle,
+  }: Props) => {
+    const api = useApi();
 
-  const isBookmarked = defined(isBookmarkedProp)
-    ? isBookmarkedProp
-    : project.isBookmarked;
+    const isBookmarked = defined(isBookmarkedProp)
+      ? isBookmarkedProp
+      : project.isBookmarked;
 
-  const toggleProjectBookmark = (event: React.MouseEvent) => {
-    update(api, {
-      orgId: organization.slug,
-      projectId: project.slug,
-      data: {isBookmarked: !isBookmarked},
-    }).catch(() => {
-      addErrorMessage(t('Unable to toggle bookmark for %s', project.slug));
-    });
+    const toggleProjectBookmark = (event: React.MouseEvent) => {
+      try {
```

Comment:
Why do we try/catch this now? I would add a comment on when this could fail if it relates to something external

Reply:
It was a `.catch` before, but I realize I need to await for that to work so, whoops